client/my-sites/stats/features/modules/stats-module-devices/stats-module-devices.tsx

```diff
@@ -0,0 +1,329 @@
+					'<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"> <path fill-rule="evenodd" clip-rule="evenodd" d="M21 12C21 16.9706 16.9706 21 12 21C9.62008 21 7.45591 20.0762 5.8465 18.5677L18.5677 5.8465C20.0762 7.45591 21 9.62008 21 12ZM5.15559 17.8444L17.8444 5.15559C16.2719 3.81158 14.2307 3 12 3C7.02944 3 3 7.02944 3 12C3 14.2307 3.81158 16.2719 5.15559 17.8444ZM22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12Z" fill="white"/> </svg>';
+		}
+
+		return {
+			id: device.key,
+			value: device.value,
+			icon,
+			name: device.label,
+			descrtipion: device.label,
+			className: `donut-${ device.key }`,
+		};
+	} );
+};
+
+const StatsModuleDevices: React.FC< StatsModuleDevicesProps > = ( {
+	path,
+	className,
+	isLoading,
+	query,
+} ) => {
+	const { devices: devicesStrings } = statsStrings();
+	const siteId = useSelector( getSelectedSiteId ) as number;
+	const translate = useTranslate();
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isNewEmptyStateEnabled = config.isEnabled( 'stats/empty-module-traffic' );
+
+	const optionLabels = {
+		[ OPTION_KEYS.SIZE ]: {
+			selectLabel: translate( 'Size' ),
+			headerLabel: translate( 'Visitors share per screen size' ),
+			analyticsId: 'screensize',
+		},
+		[ OPTION_KEYS.BROWSER ]: {
+			selectLabel: translate( 'Browser' ),
+			headerLabel: translate( 'Browser' ),
+			analyticsId: 'browser',
+		},
+		[ OPTION_KEYS.OS ]: {
+			selectLabel: translate( 'OS' ),
+			headerLabel: translate( 'Operating System' ),
+			analyticsId: 'os',
+		},
+	};
+
+	const [ selectedOption, setSelectedOption ] = useState( OPTION_KEYS.SIZE );
+	const { isFetching, data } = useModuleDevicesQuery( siteId, selectedOption, query );
+	const showLoader = isLoading || isFetching;
+
+	const changeViewButton = ( selection: SelectOptionType ) => {
+		const filter = selection.value;
+
+		// publish an event
+		const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
+		recordTracksEvent( `${ event_from }_devices_module_menu_clicked`, {
+			button: optionLabels[ filter ]?.analyticsId ?? filter,
+		} );
+
+		setSelectedOption( filter );
+	};
+
+	const toggleControlComponent = (
+		<SimplifiedSegmentedControl
+			className="stats-module-devices__tabs"
+			options={ Object.entries( optionLabels ).map( ( entry ) => ( {
+				value: entry[ 0 ], // optionLabels key
+				label: entry[ 1 ].selectLabel, // optionLabels object value
+			} ) ) }
+			initialSelected={ selectedOption }
+			// @ts-expect-error TODO: missing TS type
+			onSelect={ changeViewButton }
+		/>
+	);
+
+	const chartData = OPTION_KEYS.SIZE === selectedOption ? prepareChartData( data ) : null;
+
+	return (
+		<>
+			{ isNewEmptyStateEnabled && (
+				<>
+					{ showLoader && (
+						<StatsCardSkeleton
+							isLoading={ isFetching }
+							className={ className }
+							title={ devicesStrings.title }
+							type={ 3 }
```

Comment:
Not related to the PR: It missed my initial review, but I guess we could give the magic numbers some meaning here.

Reply:
I added a description for the property.

Reply:
In this case, it doesn't seems to be important enough, but it is still confusing to have magic numbers. Not a blocker tho.

Reply:
I can change this in future updates. What do you think would be better? "OptionA" or "Variant1" as strings instead of numbers?

Reply:
Either will be better than the magic numbers, with the naming we'd know they are just variants of the placeholder, not something else.